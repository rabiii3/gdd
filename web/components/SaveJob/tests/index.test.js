import React from 'react';
import { BrowserRouter as Router } from "react-router-dom";
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from '../../../store/configureStore.test';
import { enhance } from '..';

describe( 'app | components | SaveJobDescription | index', () => {
  it('should match snapshot', () => {
    const Component = () => <div/>;
    const EnhancedComponent = enhance(Component);
    const state = {
      skills:{
        data:{
          '5d035ef378c332001b67bada':{
            _id: '5d035ef378c332001b67bada',
            label: 'DevOps',
          },
          '5d035ef378c332001b67badb':{
            _id: '5d035ef378c332001b67badb',
            label: 'React JS',
          }
        }
      },
      jobDescription:{
        data:{
          '0':{
            _id: '0',
            title: 'required js dev',
            content: 'some content',
            skills: ['5d035ef378c332001b67bada', '5d035ef378c332001b67badb'],
            createdBy: '456',
            createdAt: '',
            updatedBy: '456',
            updatedAt: '',
            tenantId: '333'
          }
        },
        sort:{
          sortBy: 'TITLE',
          direction: 'desc',
        }
      },
    };
    const store = configureStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <Router>
         <EnhancedComponent />
        </Router>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});