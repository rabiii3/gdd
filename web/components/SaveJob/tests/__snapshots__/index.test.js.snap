// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app | components | SaveJobDescription | index should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <WithStyles(Connect(withRouter(Component)))>
        <Connect(withRouter(Component))
          classes={
            Object {
              "container": "Connect-withRouter-Component---container-1",
              "labelEdit": "Connect-withRouter-Component---labelEdit-2",
              "labelNoEdit": "Connect-withRouter-Component---labelNoEdit-3",
              "margin": "Connect-withRouter-Component---margin-4",
            }
          }
        >
          <withRouter(Component)
            addJob={[Function]}
            allSkills={
              Object {
                "5d035ef378c332001b67bada": Object {
                  "_id": "5d035ef378c332001b67bada",
                  "label": "DevOps",
                },
                "5d035ef378c332001b67badb": Object {
                  "_id": "5d035ef378c332001b67badb",
                  "label": "React JS",
                },
              }
            }
            classes={
              Object {
                "container": "Connect-withRouter-Component---container-1",
                "labelEdit": "Connect-withRouter-Component---labelEdit-2",
                "labelNoEdit": "Connect-withRouter-Component---labelNoEdit-3",
                "margin": "Connect-withRouter-Component---margin-4",
              }
            }
            skills={
              Array [
                Object {
                  "_id": "5d035ef378c332001b67bada",
                  "label": "DevOps",
                },
                Object {
                  "_id": "5d035ef378c332001b67badb",
                  "label": "React JS",
                },
              ]
            }
            updateJob={[Function]}
          >
            <Route>
              <Component
                addJob={[Function]}
                allSkills={
                  Object {
                    "5d035ef378c332001b67bada": Object {
                      "_id": "5d035ef378c332001b67bada",
                      "label": "DevOps",
                    },
                    "5d035ef378c332001b67badb": Object {
                      "_id": "5d035ef378c332001b67badb",
                      "label": "React JS",
                    },
                  }
                }
                classes={
                  Object {
                    "container": "Connect-withRouter-Component---container-1",
                    "labelEdit": "Connect-withRouter-Component---labelEdit-2",
                    "labelNoEdit": "Connect-withRouter-Component---labelNoEdit-3",
                    "margin": "Connect-withRouter-Component---margin-4",
                  }
                }
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                skills={
                  Array [
                    Object {
                      "_id": "5d035ef378c332001b67bada",
                      "label": "DevOps",
                    },
                    Object {
                      "_id": "5d035ef378c332001b67badb",
                      "label": "React JS",
                    },
                  ]
                }
                updateJob={[Function]}
              >
                <div />
              </Component>
            </Route>
          </withRouter(Component)>
        </Connect(withRouter(Component))>
      </WithStyles(Connect(withRouter(Component)))>
    </Router>
  </BrowserRouter>
</Provider>
`;
